be/src/pipeline/pipeline_fragment_context.cpp

```diff
@@ -1513,8 +1493,12 @@ Status PipelineFragmentContext::_create_operator(ObjectPool* pool, const TPlanNo
     case TPlanNodeType::SORT_NODE: {
         const auto should_spill = _runtime_state->enable_spill() &&
                                   tnode.sort_node.algorithm == TSortAlgorithm::FULL_SORT;
+        const bool use_local_merge =
+                tnode.sort_node.merge_by_exchange && _runtime_state->enable_local_merge_sort();
```

Comment:
Set `tnode.sort_node.merge_by_exchange` if `enable_local_merge_sort` is true by FE. Here we just need to decide whether to use local merge by `tnode.sort_node.merge_by_exchange`